src/vcpu.rs

```diff
@@ -31,4 +31,7 @@ pub trait VirtualCPU: Sized + Send {
 
 	/// Queries the CPUs base frequency in kHz
 	fn get_cpu_frequency(&self) -> Option<NonZeroU32>;
+
+	/// Initialize virtual CPU
+	fn init(&mut self) -> HypervisorResult<()>;
```

Comment:
I pretty much like the PR. But I'm not super happy to re-introduce a separate `init` function. I see that it might be necessary on macOS, but I'm thinking if this can be avoided.